include/ck_tile/ops/pooling/kernel/pool_kernel.hpp

```diff
@@ -0,0 +1,496 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2025, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include "ck_tile/core.hpp"
+#include "ck_tile/ops/pooling/pipeline/pool_default_policy.hpp"
+#include "ck_tile/ops/common.hpp"
+#include <type_traits>
+
+namespace ck_tile {
+
+/// @brief Host arguments for pooling operations
+template <typename TensorShape, typename WindowShape>
+struct PoolHostArgs
+{
+
+    CK_TILE_HOST PoolHostArgs(const void* input_ptr_,
+                              void* output_ptr_,
+                              TensorShape input_shape_,
+                              TensorShape output_shape_,
+                              TensorShape input_strides_,
+                              TensorShape output_strides_,
+                              WindowShape window_lengths_,
+                              WindowShape window_strides_,
+                              WindowShape window_dilations_,
+                              WindowShape input_left_pads_,
+                              WindowShape input_right_pads_)
+        : input_ptr(input_ptr_),
+          output_ptr(output_ptr_),
+          input_shape(input_shape_),
+          output_shape(output_shape_),
+          input_strides(input_strides_),
+          output_strides(output_strides_),
+          window_lengths(window_lengths_),
+          window_strides(window_strides_),
+          window_dilations(window_dilations_),
+          input_left_pads(input_left_pads_),
+          input_right_pads(input_right_pads_)
+    {
+    }
+
+    const void* input_ptr;
+    void* output_ptr;
+
+    TensorShape input_shape;
+    TensorShape output_shape;
+    TensorShape input_strides;
+    TensorShape output_strides;
+    WindowShape window_lengths;
+    WindowShape window_strides;
+    WindowShape window_dilations;
+    WindowShape input_left_pads;
+    WindowShape input_right_pads;
+};
+
+/// @brief Kernel arguments for pooling operations
+template <typename TensorShape, typename WindowShape>
+struct PoolKernelArgs
+{
+    const void* input_ptr;
+    void* output_ptr;
+    TensorShape input_shape;
+    TensorShape output_shape;
+    TensorShape input_strides;
+    TensorShape output_strides;
+    WindowShape window_lengths;
+    WindowShape window_strides;
+    WindowShape window_dilations;
+    WindowShape input_left_pads;
+    WindowShape input_right_pads;
+};
+
+template <typename Problem_, typename Policy_ = PoolDefaultPolicy>
+struct Pool
```

Comment:
the convention we follow is struct OperationKernel
```suggestion
struct PoolKernel
```